AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m

```diff
@@ -3,43 +3,129 @@
 #import "MSDataStoreInternal.h"
 #import "MSStorageIngestion.h"
 #import "MSTokensResponse.h"
+#import "MSTokenResult.h"
+#import "MSKeychainUtil.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
 static NSString *const kMSPartitions = @"partitions";
+static NSString *const kMSStorageReadOnlyDbTokenKey = @"MSStorageReadOnlyDbToken";
+static NSString *const kMSStorageUserDbTokenKey = @"MSStorageUserDbToken";
 
 @implementation MSTokenExchange : NSObject
 
-+ (void)tokenAsync:(MSStorageIngestion *)httpClient
-           partitions:(NSArray *)partitions
++ (void)performDbTokenAsyncOperationWithHttpClient:(MSStorageIngestion *)httpClient
+           partition:(NSString *)partition
     completionHandler:(MSGetTokenAsyncCompletionHandler)completion {
+    
+    MSTokenResult *cachedToken = [MSTokenExchange retrieveCachedToken:partition];
```

Comment:
nit:there is a code formatting convention on SDK:

no empty space without comment:

example:

void func(NSString *blah) {

   for (int ....) {
  }
}


to fix the above code, put a comment after the empty space(before for loop) :

void func(NSString *blah) {

  // my super important comment
   for (int ....) {
  }
}